mlir/lib/Dialect/Utils/StructuredOpsUtils.cpp

```diff
@@ -96,6 +96,79 @@ bool mlir::isRowMajorBatchMatmul(ArrayAttr indexingMaps) {
   return indexingMaps == maps;
 }
 
+bool mlir::isVecmat(ArrayAttr indexingMaps) {
+  if (indexingMaps.size() != 3)
+    return false;
+  auto map0 = cast<AffineMapAttr>(indexingMaps[0]).getValue();
```

Comment:
nit: for readability, we try to minimize the use of `auto` to some extent, leaving it only for cases where the actual type class is explicit on the right-hand side. I would use the type name here and for the mlir contexts below.